### PR TITLE
Add Zendesk to the healthcheck JSON endpoint

### PR DESCRIPTION
**Pull request by threedaymonk**
* https://www.pivotaltracker.com/story/show/96673886 (second check list item)

### CODE REVIEW

File: spec/controllers/healthcheck_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    end
+
+    it 'returns an HTTP Bad Gateway status' do
+      expect(response).to have_http_status(:bad_gateway)
+    end
+
+    it "reports #{service} as inaccessible" do
+      expect(parsed_body).to include('checks' => include(service => false))
+    end
+  end
```
Comment (SteveMarshall) [inline spec/controllers/healthcheck_controller_spec.rb]: This is a really neat use of `shared_examples` :+1:

**Comment by SteveMarshall** [comment]
Looks good to me :stars: @dan-palmer?

File: app/controllers/healthcheck_controller.rb [hunk 2]
```diff
@@ -26,4 +21,14 @@ def sendgrid_alive?
   def messagelabs_alive?
     SendgridHelper.smtp_alive?(*PrisonMailer.smtp_settings.values_at(:address, :port))
   end
+
+  def database_active?
+    ActiveRecord::Base.connection.active?
```
Comment (dan-palmer) [inline app/controllers/healthcheck_controller.rb]: I think this is the only healthcheck checking the DB that works across all of the apps in MOJ as `ActiveRecord::QueryCache` middleware has been <a href='https://github.com/ministryofjustice/prison-visits/blob/master/config/application.rb#L73'>removed</a>.

File: spec/controllers/healthcheck_controller_spec.rb (hunk 1 above)
Comment (threedaymonk) [inline spec/controllers/healthcheck_controller_spec.rb]: They were there before! I just pulled them out of the support file they were hidden in.

**Comment by dan-palmer** [comment]
Looks good to me @threedaymonk, @SteveMarshall :+1:

File: spec/controllers/healthcheck_controller_spec.rb (hunk 1 above)
Comment (SteveMarshall) [inline spec/controllers/healthcheck_controller_spec.rb]: Apparently I wrote them! Yay for memory! :dizzy_face:

File: app/controllers/healthcheck_controller.rb [hunk 3]
```diff
@@ -26,4 +21,14 @@ def sendgrid_alive?
   def messagelabs_alive?
     SendgridHelper.smtp_alive?(*PrisonMailer.smtp_settings.values_at(:address, :port))
```
Comment (dan-palmer) [inline app/controllers/healthcheck_controller.rb]: Not related to @threedaymonk's work in this PR.. I don't like the fact that what should probably be `MailerHelper` or similar is called `SendgridHelper`.

Knowing that the `PrisonMailer` isn't using Sendgrid is a little confusing. :laughing:

Comment (threedaymonk) [inline app/controllers/healthcheck_controller.rb]: I agree! I'm also not a fan of most of these 'Helper' module function things that litter the codebase.

**Comment by threedaymonk** [comment]
Tested on preprod.

**Pull request merged**